src/NuGet.Services.AzureSearch/Db2AzureSearch/Db2AzureSearchCommand.cs

```diff
@@ -0,0 +1,228 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Search;
+using Microsoft.Azure.Search.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NuGet.Services.AzureSearch.Wrappers;
+
+namespace NuGet.Services.AzureSearch.Db2AzureSearch
+{
+    public class Db2AzureSearchCommand : IDb2AzureSearchCommand
+    {
+        private readonly INewPackageRegistrationProducer _producer;
+        private readonly IIndexActionBuilder _indexActionBuilder;
+        private readonly ISearchServiceClientWrapper _serviceClient;
+        private readonly ISearchIndexClientWrapper _searchIndexClient;
+        private readonly ISearchIndexClientWrapper _hijackIndexClient;
+        private readonly IOptionsSnapshot<Db2AzureSearchConfiguration> _options;
+        private readonly ILogger<Db2AzureSearchCommand> _logger;
+
+        public Db2AzureSearchCommand(
+            INewPackageRegistrationProducer producer,
+            IIndexActionBuilder indexActionBuilder,
+            ISearchServiceClientWrapper serviceClient,
+            ISearchIndexClientWrapper searchIndexClient,
+            ISearchIndexClientWrapper hijackIndexClient,
+            IOptionsSnapshot<Db2AzureSearchConfiguration> options,
+            ILogger<Db2AzureSearchCommand> logger)
+        {
+            _producer = producer ?? throw new ArgumentNullException(nameof(producer));
+            _indexActionBuilder = indexActionBuilder ?? throw new ArgumentNullException(nameof(indexActionBuilder));
+            _serviceClient = serviceClient ?? throw new ArgumentNullException(nameof(serviceClient));
+            _searchIndexClient = searchIndexClient ?? throw new ArgumentNullException(nameof(searchIndexClient));
+            _hijackIndexClient = hijackIndexClient ?? throw new ArgumentNullException(nameof(hijackIndexClient));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task ExecuteAsync()
+        {
+            var allWork = new ConcurrentBag<NewPackageRegistration>();
+            using (var cancelledCts = new CancellationTokenSource())
+            using (var completedCts = new CancellationTokenSource())
```

Comment:
>completedCts [](start = 23, length = 12)

It's a little unclear what triggers `completedCts` from its name alone. What do you think of the name `produceWorkCts` instead of `completedCts`?